### ISSUE TITLE
Make using infinite scroll for the smart content easier on the website 

### ISSUE THREAD

**Issue by danrot** [sulu-io/sulu#1885]
If somebody wants to use the smart content in combination with infinite scroll on the website, he needs to implement an own controller and/or a special twig template.

There are a few different ways to make this easier for frontend developers:

# Introduce a smart content controller

This controller would accept URLs like `/smart-content/<page-id>/<property-name>?page=<page-no>`, and return some JSON. It would be nicer if HTML is returned, but that doesn't work, because it is not clear which template must be used for rendering.
This also means that the frontend developer has to build the same HTML structure twice (once on the server and another time using javscript for the infinite scroll).

# Using the partial flag

The partial flag renders only the content block of a page. But that is already too much in this case, although it might work. But especially if there are loads of other properties, this has a serious performance impact.

# Enable to render single properties of a template

This is like a combination of the previous two. There could be a new query param called `property`, which only renders the given property. Then we would have URLs like `/de/my/super/awesome/url?property=smartcontent&page=2`. Then the server only renders the `smartcontent` property using the other query parameters. This could be done in different ways:

* The entire template is rendered, and the required part of the response is extracted. This would be not as performant as it could be.
* We parse the twig template, remove the unnecessary parts using string manipulation, and pass an adapted string to the template engine (I think the engine also takes strings instead of files). This is also my favorite of all options.
* We make use of blocks, then the URL might change to `/de/my/super/awesome/url?block=smartcontent&page=2`. This means that the name is not coupled to the template xml configuration, but to the Twig template. Would maybe add some more flexibility, but the frontend developer must take care of this.

**Comment by wachterjohannes** [comment]
if we are not rendering the whole template this is very hard to acchieve and it depends on the template a lot ... if you use a variable which will be defined outside the rendered block the rendering will crash! if we want to return parts of the html we should use the same functionality like preview which is not very performant but works pretty fine.

**Comment by patkar** [comment]
I think a developer which wants to use the same template/twig partial for initial rendering and as single property via Ajax simply must be aware of using only the context which is available in the partial property rendering.

**Comment by danrot** [comment]
@wachterjohannes I agree with @patkar, it's the same as with the preview, you have to implement that stuff properly if you want to use that feature. And at least there would be a clear error message.

And apart from that I don't know what you are actually talking about at all :smiley: According to the [Twig Documentation](http://twig.sensiolabs.org/doc/tags/extends.html#block-nesting-and-scope) it is possible to use outer scope variables in blocks.

**Comment by wachterjohannes** [comment]
i cannot think that this can work (: but yeah we have to document this properly than it would be easy to implement!

**Comment by niklasnatter** [comment]
This code snippet might be helpful if you are looking into implementing a `Load more` button for a `smart_content` property in your project: https://gist.github.com/luca-rath/d16c0ad74fa06da49fed22404a53398b
